app/models/host.rb

```diff
@@ -18,6 +18,15 @@ class Host < ApplicationRecord
   has_many :policies, through: :policy_hosts
   has_many :assigned_profiles, through: :policies, source: :profiles
 
+  def self.os_versions(hosts)
+    select(:system_profile).where(id: hosts).pluck(
+      %w[
+        system_profile->'operating_system'->'major'
+        system_profile->'operating_system'->'minor'
+      ]
+    ).map { |version| version.join('.') }.uniq
```

Comment:
It'd be great to use `distinct` instead of `uniq`. You may need to store the array in `pluck` in a constant, since we'd need to use it twice:

```rb
Host.select(%w(system_profile->'operating_system'->'major' system_profile->'operating_system'->'minor')).distinct.pluck(%w(system_profile->'operating_system'->'major' system_profile->'operating_system'->'minor'))
```

resulting query:

```sql
SELECT DISTINCT system_profile->'operating_system'->'major', system_profile->'operating_system'->'minor' FROM "inventory"."hosts"
```

Reply:
I just noticed that the `system_profile` is json-based :scream: last time (2013?) I tried to use this in rails, it was crashing :laughing: